syntax = "proto3";

package hello;

service SuggestionsService {
    rpc Suggest (SuggestionRequest) returns (SuggestionResponse);
    rpc VectorClockUpdate (VectorClockInp_sugg) returns (Empty_sugg);
}

message Empty_sugg {}

message VectorClockInp_sugg {
    repeated int32 vector_clock = 1;
}

message Book {
    string bookId = 1;
    string title = 2;
    string author = 3;
}

message SuggestionRequest {
    repeated Book books = 1;
    repeated string ordered = 2;
}

message SuggestionResponse {
    repeated Book books = 1;
}
